config/config.go

```diff
@@ -983,9 +932,35 @@ func (p PackageConfig) Equals(other PackageConfig) bool {
 	return reflect.DeepEqual(p, other)
 }
 
-// PackageReference contains the deconstructed parts of a package reference in the config.
-// Eg: ${packages.some-package}/path/a/b/c -> {"some-package", "/path/a/b/c"}.
-type PackageReference struct {
-	Package       string
-	PathInPackage string
+// LocalDataParentDirectory returns the folder that will contain the all packages of this type.
+// Ex: /home/user/.viam/packages/ml_models
+func (p *PackageConfig) LocalDataParentDirectory() string {
```

Comment:
nit: rename this to be localParentDirectory if we are not using .data anymore

Reply:
I think I just made the name even worse because we brought back `.data` ;/ but also now it accepts a parent dir